src/pages/[productSlug]/integrations/[integrationSlug]/[integrationVersion]/components/[componentSlug]/index.tsx

```diff
@@ -0,0 +1,162 @@
+import { cachedGetProductData } from 'lib/get-product-data'
+import {
+	Integration,
+	fetchIntegration,
+} from 'lib/integrations-api-client/integration'
+import {
+	Release,
+	ReleaseComponent,
+	fetchIntegrationRelease,
+} from 'lib/integrations-api-client/release'
+import serializeIntegrationMarkdown from 'lib/serialize-integration-markdown'
+import { withTiming } from 'lib/with-timing'
+import { ProductData, ProductSlug } from 'types/products'
+import ProductIntegrationComponentView from 'views/product-integration/component-view'
+import { integrationVersionBreadcrumbLinks } from '../..'
+
+interface PathParams {
+	productSlug: ProductSlug
+	integrationSlug: string
+	integrationVersion: string
+	componentSlug: string
+}
+
+async function _getServerSideProps({
+	params: { productSlug, integrationSlug, integrationVersion, componentSlug },
+}: {
+	params: PathParams
+}) {
+	// Pull out the Product Config
+	const product = cachedGetProductData(productSlug)
+
+	// 404 early if the product is not enabled for integrations
+	if (!ENABLED_INTEGRATION_PRODUCTS.includes(product.slug)) {
+		return {
+			notFound: true,
+		}
+	}
+
+	// Fetch the Integration
+	const integrationResponse = await fetchIntegration(
+		product.slug,
+		integrationSlug
+	)
+	if (integrationResponse.meta.status_code != 200) {
+		console.warn('Could not fetch Integration', integrationResponse)
+		return { notFound: true }
+	}
+	const integration = integrationResponse.result
+
+	// If the integration is external only, we shouldn't render this page
+	if (integration.external_only) {
+		return {
+			notFound: true,
+		}
+	}
+
+	// Fetch the Release
+	const activeReleaseResponse = await fetchIntegrationRelease(
+		product.slug,
+		integrationSlug,
+		integrationVersion === 'latest'
+			? integration.versions[0]
+			: integrationVersion
+	)
+	if (activeReleaseResponse.meta.status_code != 200) {
+		console.warn('Could not fetch Release', activeReleaseResponse)
+		return { notFound: true }
+	}
+	const activeRelease = activeReleaseResponse.result
+
+	// Grab the specific release component
+	const releaseComponent: ReleaseComponent = activeRelease.components
+		.filter((rc: ReleaseComponent) => {
+			// Only release components w/ either a README or variable groups get a page
+			return rc.readme || rc.variable_groups.length
+		})
+		.find((rc: ReleaseComponent) => {
+			return rc.component.slug === componentSlug
+		})
+	if (!releaseComponent) {
+		return { notFound: true }
+	}
+
+	// If the integration version is the latest version, redirect using `latest`
+	if (integrationVersion === integration.versions[0]) {
+		return {
+			redirect: {
+				destination: `/${product.slug}/integrations/${integration.slug}/latest/components/${releaseComponent.component.slug}`,
+				// Not permanent as a new release in the future will turn the
+				// latest release into an older release which should render!
+				permanent: false,
+			},
+		}
+	}
```

Comment:
So we're always redirecting `<latest_version>` to `latest`

I wonder if we actually want this the other way around?

---

Idea comes from
- https://github.com/hashicorp/waypoint/releases/latest
  -  which redirects to https://github.com/hashicorp/waypoint/releases/tag/v0.10.5

Reply:
Oh interesting..

I was thinking we keep `latest` stable so the SEO juices flow into a singular place consistently.

Reply:
I don't know if that is a sound decision though.

Reply:
Thinking about this some more out loud:

Looking at the GitHub context, I think that GitHub would want each version to be identified as **a totally new thing with different content**, as each release will have notes on what was changed from the previous release (and it's not like something that's being slowly incrementally updated.  That being said, I think it makes sense to have *each release* be its own page.

Our use case is slightly different, as our `releases` are more like git commits, slowly incrementing the content over time.  I think our use case here is closer to how our default docs pages latest version get a URL without the version in it (e.g. https://developer.hashicorp.com/packer/docs/templates, vs https://developer.hashicorp.com/packer/docs/v1.7.x/templates).

If we included the version in our docs, I believe that our SEO would be implicated each time we cut a new version.  (Maybe that could be avoided if we used a canonical URL pointing to the new place), but I think it might confuse google into thinking it's a separate document.

Reply:
I also think we should keep stable URLs (ex. `/waypoint/integrations/docker/0.10.3/components/builder`) stable.

Vanity urls (`/waypoint/integrations/docker` or `/waypoint/integrations/docker/latest`) can have the underlying content/data shift.

Reply:
> If we included the version in our docs, I believe that our SEO would be implicated each time we cut a new version. (Maybe that could be avoided if we used a canonical URL pointing to the new place), but I think it might confuse google into thinking it's a separate document.

Hmm, I cannot say I know what Google/Crawlers do in this case `:homer:`, so I defer to your call!

Reply:
I can't say for certain either TBH!  Let's discuss a bit more after you wrap up your review on this!